lib/rubocop/cop/rspec/rails/inferred_spec_type.rb

```diff
@@ -0,0 +1,127 @@
+            { prefix: 'spec/controllers/', type: :controller },
+            { prefix: 'spec/features/', type: :feature },
+            { prefix: 'spec/generator/', type: :generator },
+            { prefix: 'spec/helpers/', type: :helper },
+            { prefix: 'spec/jobs/', type: :job },
+            { prefix: 'spec/mailboxes/', type: :mailbox },
+            { prefix: 'spec/mailers/', type: :mailer },
+            { prefix: 'spec/models/', type: :model },
+            { prefix: 'spec/requests/', type: :request },
+            { prefix: 'spec/integration/', type: :request },
+            { prefix: 'spec/api/', type: :request },
+            { prefix: 'spec/routing/', type: :routing },
+            { prefix: 'spec/system/', type: :system },
+            { prefix: 'spec/views/', type: :view }
+          ].freeze
+
+          MSG = 'Remove redundant spec type.'
+
+          RESTRICT_ON_SEND = %i[
+            describe
+          ].freeze
+
+          # @param [RuboCop::AST::SendNode] node
+          def on_send(node)
+            pair_node = describe_with_type(node)
+            return unless pair_node
+            return unless inferred_type?(pair_node)
+
+            removable_node = detect_removable_node(pair_node)
+            add_offense(removable_node) do |corrector|
+              autocorrect(corrector, removable_node)
+            end
+          end
+
+          private
+
+          # @!method describe_with_type(node)
+          #   @param [RuboCop::AST::SendNode] node
+          #   @return [RuboCop::AST::PairNode, nil]
+          def_node_matcher :describe_with_type, <<~PATTERN
+            (send
+              { (const nil? :RSpec) | nil? }
+              _
+              _
+              _*
+              ({ hash | kwargs }
+                (pair ...)*
+                $(pair (sym :type) (sym _))
+                (pair ...)*
+              )
+            )
+          PATTERN
+
+          # @param [RuboCop::AST::Corrector] corrector
+          # @param [RuboCop::AST::Node] node
+          def autocorrect(corrector, node)
+            corrector.remove(
+              node.location.expression.with(
+                begin_pos: node.left_sibling.location.expression.end_pos
+              )
+            )
+          end
+
+          # @param [RuboCop::AST::PairNode] node
+          # @return [RuboCop::AST::Node]
+          def detect_removable_node(node)
+            if node.parent.pairs.size == 1
+              node.parent
+            else
+              node
+            end
+          end
+
+          # @return [String]
+          def file_path
+            processed_source.file_path
+          end
+
+          # @param [RuboCop::AST::PairNode] node
+          # @return [Boolean]
+          def inferred_type?(node)
+            inferred_type_from_file_path.inspect == node.value.source
+          end
+
+          # @return [Symbol, nil]
+          def inferred_type_from_file_path
+            INFERENCES.find do |inference|
```

Comment:
Would it be possible and safe to use `RSpec::Rails::DIRECTORY_MAPPINGS` instead? ([ref](https://github.com/rspec/rspec-rails/blob/6b4969ef8e20cbd21c51840c158b91f90630041c/lib/rspec/rails/configuration.rb#L28)).

Reply:
Indeed, I wrote this definition based on that part of rspec-rails. 
By "use `RSpec::Rails::DIRECTORY_MAPPINGS`" do you mean that we should `require "rspec-rails"` and use it here? Or does it mean that we should try to define a Hash of the same format?

Reply:
> do you mean that we should `require "rspec-rails"` and use it here?

Yes, much like [how the Rails/HttpStatus cop is loaded](https://github.com/rubocop/rubocop-rspec/blob/master/lib/rubocop/cop/rspec_cops.rb#L16-L20).

Perhaps first we should hear the opinion of @rubocop/rubocop-rspec if it’s a good idea.

Reply:
Directory mappings can be reconfigured in `RSpec.configure`, and using the defaults will break the principle of the least surprise. I don't think there's a good way of loading RSpec configuration in RuboCop's runtime. E.g. we ended up duplicating aliases configuration in `RSpec/Language` in `.rubocop.yml`.

There's also an `@api private` note in RSpec Rails, meaning it can change any moment. And we would have to support that. And reasonably, we can't add an explicit runtime dependency constraint on specific RSpec Rails versions, so once RSpec Rails change it, this will might stop working properly.

Reply:
Could we offload the configuration to our users? And then simply copy-paste the default values into config/default.yml?